### DISCUSSION TITLE
Add inner join example to reference docs

### DISCUSSION THREAD

**Discussion by gregnr** [supabase/supabase#16846] [Contribute to Supabase]
# Improve documentation

## Link 

Each of these client lib reference docs:

- https://supabase.com/docs/reference/javascript/select
- https://supabase.com/docs/reference/python/select
- https://supabase.com/docs/reference/swift/select
- https://supabase.com/docs/reference/kotlin/select

## Describe the improvement

These should include an example on how to do `inner` joins as the Dart and C# docs do:

- https://supabase.com/docs/reference/dart/select
![image](https://github.com/supabase/supabase/assets/4133076/ef614639-e111-49de-a386-4d6c70cb8a4f)

- https://supabase.com/docs/reference/csharp/select
![image](https://github.com/supabase/supabase/assets/4133076/aa8533cf-93de-4fc7-a6ba-2dbf8ce2ccc3)

## Additional context

Examples are defined under each `./spec/supabase_<lang>_<version>.yml` file. Eg. JavaScript:
https://github.com/supabase/supabase/blob/843d4e01ebec050824be0c56275a5b9a251d3e01/spec/supabase_js_v2.yml#L1129

Let's also change the "Filtering through foreign tables" examples to use `inner`.

**Comment by SamantaTarun** [comment]
I'd like to work on it.

**Comment by gregnr** [comment]
@tarunsamanta2k20 that'd be great!

**Comment by SamantaTarun** [comment]
How do I run the query after preparing to test if it works?

**Comment by SamantaTarun** [comment]
Also if you have any references for writing language specific query then please mention the link over here.

**Comment by gregnr** [comment]
Hey @tarunsamanta2k20, feel free to follow our guide on creating API routes and querying them (specifically under "REST API"):
https://supabase.com/docs/guides/api/creating-routes

This should help you get started for JavaScript, but the same logic will apply to other languages. Also each language includes install instructions:
https://supabase.com/docs/reference/javascript/installing

**Comment by danielbolivar** [comment]
I would like to know if this issue has been solved. If not, may I work on it?

**Comment by gregnr** [comment]
This issue is still open!

**Comment by bk7312** [comment]
I encountered a similar issue (in JS) when trying to filter via a foreign table, the docs mentioned something about using `!inner` and it wasn't obvious how to do it. Managed to figure it out and made a PR https://github.com/supabase/supabase/pull/17387 to add an example. 

Also, it seems like @danielbolivar's PR https://github.com/supabase/supabase/pull/15438 is still open?
